graph-manager-docs/source/federation.mdx

```diff
@@ -127,9 +127,113 @@ server.listen().then(({ url }) => {
+to other modifications. This helps avoid the scenario where the query planner generates queries that fail validation in downstream services or violate
+resolvers' expectations. Some examples of this include:
+
+1. Modifying `@key` or `@requires` dependencies while simultaneously modifying dependency fields
+2. Removing a type implementation from an interface
+
+For example, let's say we define a Channel interface in one service and define types that implement that interface in two other services:
+
+```graphql
+# Channel service
+interface Channel @key(fields: "id") {
+  id: ID!
+}
+
+# Web service
+type WebChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+
+# Email service
+type EmailChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+```
+
+To remove the `EmailChannel` implementation of the `Channel` interface:
+
+1. Register a new `email` service that removes the `EmailChannel` implementation of `Channel`.
+1. Deploy a new Email service that removes the implementation.
+
+The first step causes the query planner to stop sending along fragments `...on EmailChannel`, which would fail validation if sent
+to a service that isn't aware of that type.
+
+#### Modifying query-planning logic
+
+You should treat migrations of query-planning logic similarly to how you treat database migrations, by carefully considering the effects on the
+downstream services as the query planner changes, and planning for "double reading" as appropriate. For example, consider a migration of deprecating
+and replacing a field in an entity which another service `@requires`. Specifically, we'll consider the following example of a `Products` service
+and `Reviews` service:
+
+```graphql
+# Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String!
+}
+
+# Review Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  reviews: [Review]! @requires(fields: "nameLowercase")
+  nameLowercase: String! @external
+}
+```
+
+If we want to deprecate the `nameLowercase` field and replace it with the `name` field:
+
+```graphql
+# Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String! @deprecated
+  name: String!
+}
+
+# Reviews Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  nameLowercase: String! @external
+  name: String! @external
+  reviews: [Review]! @requires(fields: "name")
+}
+```
+
+To do this migration in-place:
+
+1. Modify the `Products` service to add the new field. (As in most cases, first deploy all replicas, then `apollo service:push` the new partial schema.)
+1. Deploy a new version of the `Reviews` service with a resolver that accepts _either_ `nameLowercase` or `name` in the source object.
+1. Modify the Reviews service's schema in the registry so that it @requires the new `name` field.
```

Comment:
Missing typewriter text on `@requires`.

Probably should just spell out `@requires(fields: "name")` though (which also makes clear that we're not ADDING name, we're REPLACING).

Reply:
thanks!